protocols/shell-workspace-creation-v1.xml

```diff
@@ -0,0 +1,321 @@
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_arbitrary_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_arbitrary_target">
+      <description summary="create a workspace target relative to another workspace">
+        This requests creates a workspace target that targets an arbitrary
+        compositor chosen name and location for the workspace.
+      </description>
+
+      <arg name="descriptor" type="new_id" interface="shell_workspace_creation_target_descriptor_v1"/>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_coordinate_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_coordinate_target">
+      <description summary="create a workspace target at the given coordinates">
+        This request creates a workspace target at the given coordinates in
+        the compositor's workspace coordinate space. This coordinate space
+        should match the one used in 'ext_workspace_handle_v1.coordinates'.
+
+        A given set of coordinates may be unrepresentable or otherwise disallowed
+        by the compositor, in which case the workspace target will be invalid.
+
+        If a workspace already exists at the given coordinates, the resuling
+        target should be valid and refer to the existing workspace.
+      </description>
+
+      <arg name="descriptor" type="new_id" interface="shell_workspace_creation_target_descriptor_v1"/>
+      <arg name="coordinates" type="array"/>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_relative_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_relative_target">
+      <description summary="create a workspace target relative to another workspace">
+        This request creates a workspace target at the given coordinate offset
+        from another workspace. The workspace coordinate space should match the
+        one used in 'ext_workspace_handle_v1.coordinates'.
+
+        A given coordinate offset may be unrepresentable or otherwise disallowed
+        by the compositor, in which case the workspace target will be invalid.
+
+        If a workspace already exists at the given offset, the resuling
+        target should be valid and refer to the existing workspace.
```

Comment:
Concern about this and to some extent get_coordinate_target. Presumably, the intention here was to allow "insert new workspace below/above nth" for dynamic workspace systems. However, as written, this can only target "above 0" or "below last" workspace: anything in the middle + offset=1 will just target the next/previous existing workspace.

Reply:
That was the intent, but I see the problem. Would an "insert at +1/-1" request work better?

In that case we'd interpret that as +1/-1 at time of object creation. E.g. if you were to create two +1 relative targets and use them sequentially, both would point to the same workspace.

Reply:
I guess it would work better yeah